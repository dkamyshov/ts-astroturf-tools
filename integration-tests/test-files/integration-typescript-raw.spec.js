const path = require('path');
const runner = require('./runner');
const fs = require('fs');
const testlist = require('../projects/integration-typescript-raw/testlist');

jest.setTimeout(30000);

describe('typescript-raw', () => {
  const workingDirectory = path.resolve(
    __dirname,
    '../projects/integration-typescript-raw'
  );

  beforeAll(async () => {
    await runner.run(workingDirectory, 'yarn', []);
  });

  testlist.forEach(filename => {
    it(`css snapshot test - ${filename}`, async () => {
      const resultCSSFileContent = fs
        .readFileSync(path.resolve(workingDirectory, `lib/${filename}.js`))
        .toString();

      expect(resultCSSFileContent).toMatchSnapshot();
    });
  });
});
