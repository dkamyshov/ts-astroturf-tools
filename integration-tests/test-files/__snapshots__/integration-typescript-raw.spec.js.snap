// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typescript-raw css snapshot test - styled-remote-interpolation 1`] = `
"import styled from 'astroturf';
import { RED, GREEN } from './utils/colors';
export const SomeComponent = styled.div \`
  color: \${\\"red\\"};
  background: \${\\"green\\"};
\`;
//# sourceMappingURL=styled-remote-interpolation.js.map"
`;

exports[`typescript-raw css snapshot test - xcss 1`] = `
"import { xcss } from '../../../../xcss';
export const someClassName = (css \`
.someClassName {

  color: red;

}\`).someClassName;
//# sourceMappingURL=xcss.js.map"
`;

exports[`typescript-raw css snapshot test - xcss-local-interpolation 1`] = `
"import { xcss } from '../../../../xcss';
const RED = 'red';
export const someClassName = (css \`
.someClassName {

  color: \${RED};

}
\`).someClassName;
//# sourceMappingURL=xcss-local-interpolation.js.map"
`;

exports[`typescript-raw css snapshot test - xcss-property-assignment 1`] = `
"import { xcss } from '../../../../xcss';
export const classes = {
    someClassName: (function () {
        var p_someClassName = css \`
.someClassName {

    color: red;
  
}\`;
        return p_someClassName.someClassName;
    })(),
};
//# sourceMappingURL=xcss-property-assignment.js.map"
`;

exports[`typescript-raw css snapshot test - xcss-property-assignment-local-interpolation 1`] = `
"import { xcss } from '../../../../xcss';
const RED = 'red';
export const classes = {
    someClassName: (function () {
        var p_someClassName = css \`
.someClassName {

    color: \${RED};
  
}
\`;
        return p_someClassName.someClassName;
    })(),
};
//# sourceMappingURL=xcss-property-assignment-local-interpolation.js.map"
`;

exports[`typescript-raw css snapshot test - xcss-property-assignment-remote-interpolation 1`] = `
"import { RED, GREEN } from './utils/colors';
import { xcss } from '../../../../xcss';
export const classes = {
    someClassName: (function () {
        var p_someClassName = css \`
.someClassName {

    color: \${\\"red\\"};
    background: \${\\"green\\"};
  
}
\`;
        return p_someClassName.someClassName;
    })(),
};
//# sourceMappingURL=xcss-property-assignment-remote-interpolation.js.map"
`;

exports[`typescript-raw css snapshot test - xcss-remote-interpolation 1`] = `
"import { RED, GREEN } from './utils/colors';
import { xcss } from '../../../../xcss';
export const someClassName = (css \`
.someClassName {

  color: \${\\"red\\"};
  background: \${\\"green\\"};

}
\`).someClassName;
//# sourceMappingURL=xcss-remote-interpolation.js.map"
`;
